src/Controls/src/Core/NavigationModel.cs

```diff
@@ -91,9 +91,24 @@ public Page PopModal()
 		{
 			if (_navTree.Count <= 1)
 				throw new InvalidNavigationException("Can't pop modal without any modals pushed");
+
+			var previousPage = CurrentPage;
 			Page modal = _navTree.Last()[0];
 			_modalStack.Remove(modal);
 			_navTree.Remove(_navTree.Last());
+
+			// Shell handles its own page life cycle events
+			// because you can pop multiple pages in a single
+			// request
+			if (_navTree.Count > 0 && 
```

Comment:
Should include protections when accessing these indexes?

Reply:
These protections probably aren't needed at all, especially given the surrounding code and the check at the top, but it just seemed safer to include them. Ideally this method should be changed to support an array of modals being pushed/popped but that would require a good amount of refactoring that's out of scope for the intent of this PR.